### ISSUE TITLE
Crash report version 6.1.2

### ISSUE THREAD

**Issue by jdmarshall90** [patchthecode/JTAppleCalendar#340]
I am using v6.1.2, and I have received a crash report. The only other information besides this report is that this happened on relaunching my app (my calendar view is on the first screen).

The offending line appears to be (though I don't see how that is possible):

```swift
let reuseIdentifier: String
```

```
# Crashlytics - plaintext stacktrace downloaded by REDACTED at Fri, 10 Mar 2017 01:55:45 GMT
# URL: 
# Organization: 
# Platform: ios
# Application:
# Version: N/A
# Bundle Identifier:
# Issue #: 18
# Issue ID: 
# Session ID: 
# Date: 2017-03-09T10:00:19Z
# OS Version: 10.2.1 (14D27)
# Device: iPhone SE
# RAM Free: 4%
# Disk Free: 48.3%

#0. Crashed: com.apple.main-thread
0  JTAppleCalendar                0x1002e1a30 specialized JTAppleCalendarView.collectionView(UICollectionView, cellForItemAt : IndexPath) -> UICollectionViewCell (UICollectionViewDelegates.swift:17)
1  JTAppleCalendar                0x1002e0080 @objc JTAppleCalendarView.collectionView(UICollectionView, cellForItemAt : IndexPath) -> UICollectionViewCell (UICollectionViewDelegates.swift)
2  UIKit                          0x18cfebc0c -[UICollectionView _createPreparedCellForItemAtIndexPath:withLayoutAttributes:applyAttributes:isFocused:notify:] + 448
3  UIKit                          0x18c763b54 -[UICollectionView _updateVisibleCellsNow:] + 3828
4  UIKit                          0x18c75e5a8 -[UICollectionView layoutSubviews] + 288
5  UIKit                          0x18c6ffa80 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1196
6  QuartzCore                     0x189bad9d8 -[CALayer layoutSublayers] + 148
7  QuartzCore                     0x189ba24cc CA::Layer::layout_if_needed(CA::Transaction*) + 292
8  UIKit                          0x18c714500 -[UIView(Hierarchy) layoutBelowIfNeeded] + 1016
9  UIKit                          0x18c7bc720 -[UINavigationController _layoutViewController:] + 1196
10 UIKit                          0x18c7ba004 -[UINavigationController _layoutTopViewController] + 228
11 UIKit                          0x18c7d2f54 -[UINavigationController navigationTransitionView:didEndTransition:fromView:toView:] + 760
12 UIKit                          0x18c7d2c20 -[UINavigationTransitionView _notifyDelegateTransitionDidStopWithContext:] + 420
13 UIKit                          0x18c7d2878 -[UINavigationTransitionView _cleanupTransition] + 724
14 UIKit                          0x18c73b0dc -[UIViewAnimationState sendDelegateAnimationDidStop:finished:] + 312
15 UIKit                          0x18c739294 +[UIViewAnimationState popAnimationState] + 324
16 UIKit                          0x18c7c610c -[UINavigationTransitionView transition:fromView:toView:] + 1972
17 UIKit                          0x18c7bbe5c -[UINavigationController _startTransition:fromViewController:toViewController:] + 2572
18 UIKit                          0x18c7baf28 -[UINavigationController _startDeferredTransitionIfNeeded:] + 856
19 UIKit                          0x18c7baadc -[UINavigationController __viewWillLayoutSubviews] + 64
20 UIKit                          0x18c7baa40 -[UILayoutContainerView layoutSubviews] + 188
21 UIKit                          0x18c6ffa80 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1196
22 QuartzCore                     0x189bad9d8 -[CALayer layoutSublayers] + 148
23 QuartzCore                     0x189ba24cc CA::Layer::layout_if_needed(CA::Transaction*) + 292
24 QuartzCore                     0x189ba238c CA::Layer::layout_and_display_if_needed(CA::Transaction*) + 32
25 QuartzCore                     0x189b1f3e0 CA::Context::commit_transaction(CA::Transaction*) + 252
26 QuartzCore                     0x189b46a68 CA::Transaction::commit() + 512
27 QuartzCore                     0x189b47488 CA::Transaction::observer_callback(__CFRunLoopObserver*, unsigned long, void*) + 120
28 CoreFoundation                 0x1867fa0c0 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 32
29 CoreFoundation                 0x1867f7cf0 __CFRunLoopDoObservers + 372
30 CoreFoundation                 0x1867262d8 CFRunLoopRunSpecific + 476
31 UIKit                          0x18c76d7b0 -[UIApplication _run] + 608
32 UIKit                          0x18c768534 UIApplicationMain + 208
33 MY_APP_NAME                    0x1000931e0 main (MY_APP_NAMEDateTimePickerView.swift:15)
34 libdispatch.dylib              0x1857095b8 (Missing)

--

#0. Crashed: com.apple.main-thread
0  JTAppleCalendar                0x1002e1a30 specialized JTAppleCalendarView.collectionView(UICollectionView, cellForItemAt : IndexPath) -> UICollectionViewCell (UICollectionViewDelegates.swift:17)
1  JTAppleCalendar                0x1002e0080 @objc JTAppleCalendarView.collectionView(UICollectionView, cellForItemAt : IndexPath) -> UICollectionViewCell (UICollectionViewDelegates.swift)
2  UIKit                          0x18cfebc0c -[UICollectionView _createPreparedCellForItemAtIndexPath:withLayoutAttributes:applyAttributes:isFocused:notify:] + 448
3  UIKit                          0x18c763b54 -[UICollectionView _updateVisibleCellsNow:] + 3828
4  UIKit                          0x18c75e5a8 -[UICollectionView layoutSubviews] + 288
5  UIKit                          0x18c6ffa80 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1196
6  QuartzCore                     0x189bad9d8 -[CALayer layoutSublayers] + 148
7  QuartzCore                     0x189ba24cc CA::Layer::layout_if_needed(CA::Transaction*) + 292
8  UIKit                          0x18c714500 -[UIView(Hierarchy) layoutBelowIfNeeded] + 1016
9  UIKit                          0x18c7bc720 -[UINavigationController _layoutViewController:] + 1196
10 UIKit                          0x18c7ba004 -[UINavigationController _layoutTopViewController] + 228
11 UIKit                          0x18c7d2f54 -[UINavigationController navigationTransitionView:didEndTransition:fromView:toView:] + 760
12 UIKit                          0x18c7d2c20 -[UINavigationTransitionView _notifyDelegateTransitionDidStopWithContext:] + 420
13 UIKit                          0x18c7d2878 -[UINavigationTransitionView _cleanupTransition] + 724
14 UIKit                          0x18c73b0dc -[UIViewAnimationState sendDelegateAnimationDidStop:finished:] + 312
15 UIKit                          0x18c739294 +[UIViewAnimationState popAnimationState] + 324
16 UIKit                          0x18c7c610c -[UINavigationTransitionView transition:fromView:toView:] + 1972
17 UIKit                          0x18c7bbe5c -[UINavigationController _startTransition:fromViewController:toViewController:] + 2572
18 UIKit                          0x18c7baf28 -[UINavigationController _startDeferredTransitionIfNeeded:] + 856
19 UIKit                          0x18c7baadc -[UINavigationController __viewWillLayoutSubviews] + 64
20 UIKit                          0x18c7baa40 -[UILayoutContainerView layoutSubviews] + 188
21 UIKit                          0x18c6ffa80 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1196
22 QuartzCore                     0x189bad9d8 -[CALayer layoutSublayers] + 148
23 QuartzCore                     0x189ba24cc CA::Layer::layout_if_needed(CA::Transaction*) + 292
24 QuartzCore                     0x189ba238c CA::Layer::layout_and_display_if_needed(CA::Transaction*) + 32
25 QuartzCore                     0x189b1f3e0 CA::Context::commit_transaction(CA::Transaction*) + 252
26 QuartzCore                     0x189b46a68 CA::Transaction::commit() + 512
27 QuartzCore                     0x189b47488 CA::Transaction::observer_callback(__CFRunLoopObserver*, unsigned long, void*) + 120
28 CoreFoundation                 0x1867fa0c0 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 32
29 CoreFoundation                 0x1867f7cf0 __CFRunLoopDoObservers + 372
30 CoreFoundation                 0x1867262d8 CFRunLoopRunSpecific + 476
31 UIKit                          0x18c76d7b0 -[UIApplication _run] + 608
32 UIKit                          0x18c768534 UIApplicationMain + 208
33 MY_APP_NAME                     0x1000931e0 main (MY_APP_NAMEDateTimePickerView.swift:15)
34 libdispatch.dylib              0x1857095b8 (Missing)

#1. Thread
0  libsystem_kernel.dylib         0x18581ba88 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1858df344 _pthread_wqthread + 1452
2  libsystem_pthread.dylib        0x1858ded8c start_wqthread + 4

#2. Thread
0  libsystem_kernel.dylib         0x18581ba88 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1858df344 _pthread_wqthread + 1452
2  libsystem_pthread.dylib        0x1858ded8c start_wqthread + 4

#3. Thread
0  libsystem_kernel.dylib         0x18581ba88 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1858df160 _pthread_wqthread + 968
2  libsystem_pthread.dylib        0x1858ded8c start_wqthread + 4

#4. Thread
0  libsystem_kernel.dylib         0x18581ba88 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1858df344 _pthread_wqthread + 1452
2  libsystem_pthread.dylib        0x1858ded8c start_wqthread + 4

#5. Thread
0  libsystem_kernel.dylib         0x18581ba88 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1858df344 _pthread_wqthread + 1452
2  libsystem_pthread.dylib        0x1858ded8c start_wqthread + 4

#6. com.apple.uikit.eventfetch-thread
0  libsystem_kernel.dylib         0x1857fd188 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1857fcff8 mach_msg + 72
2  CoreFoundation                 0x1867fa5d0 __CFRunLoopServiceMachPort + 192
3  CoreFoundation                 0x1867f81ec __CFRunLoopRun + 1132
4  CoreFoundation                 0x1867262b8 CFRunLoopRunSpecific + 444
5  Foundation                     0x18726326c -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 304
6  Foundation                     0x187283dd0 -[NSRunLoop(NSRunLoop) runUntilDate:] + 96
7  UIKit                          0x18d0e1c38 -[UIEventFetcher threadMain] + 136
8  Foundation                     0x187360e68 __NSThread__start__ + 1024
9  libsystem_pthread.dylib        0x1858e1850 <redacted> + 240
10 libsystem_pthread.dylib        0x1858e1760 _pthread_start + 282
11 libsystem_pthread.dylib        0x1858ded94 thread_start + 4

#7. com.twitter.crashlytics.ios.MachExceptionServer
0  MY_APP_NAME                     0x1000f3788 CLSProcessRecordAllThreads (CLSProcess.c:376)
1  MY_APP_NAME                     0x1000f3788 CLSProcessRecordAllThreads (CLSProcess.c:376)
2  MY_APP_NAME                     0x1000f3644 CLSProcessRecordAllThreads (CLSProcess.c:407)
3  MY_APP_NAME                     0x1000e3df0 CLSHandler (CLSHandler.m:26)
4  MY_APP_NAME                     0x1000ded94 CLSMachExceptionServer (CLSMachException.c:446)
5  libsystem_pthread.dylib        0x1858e1850 <redacted> + 240
6  libsystem_pthread.dylib        0x1858e1760 _pthread_start + 282
7  libsystem_pthread.dylib        0x1858ded94 thread_start + 4

#8. Thread
0  libsystem_kernel.dylib         0x18581ba88 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1858df344 _pthread_wqthread + 1452
2  libsystem_pthread.dylib        0x1858ded8c start_wqthread + 4

#9. Thread
0  libsystem_kernel.dylib         0x18581ba88 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1858df344 _pthread_wqthread + 1452
2  libsystem_pthread.dylib        0x1858ded8c start_wqthread + 4

#10. Thread
0  libsystem_kernel.dylib         0x18581ba88 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1858df344 _pthread_wqthread + 1452
2  libsystem_pthread.dylib        0x1858ded8c start_wqthread + 4

#11. Thread
0  libsystem_kernel.dylib         0x18581ba88 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1858df344 _pthread_wqthread + 1452
2  libsystem_pthread.dylib        0x1858ded8c start_wqthread + 4

#12. com.apple.NSURLConnectionLoader
0  libsystem_kernel.dylib         0x1857fd188 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1857fcff8 mach_msg + 72
2  CoreFoundation                 0x1867fa5d0 __CFRunLoopServiceMachPort + 192
3  CoreFoundation                 0x1867f81ec __CFRunLoopRun + 1132
4  CoreFoundation                 0x1867262b8 CFRunLoopRunSpecific + 444
5  CFNetwork                      0x186f2ba70 <redacted> + 336
6  Foundation                     0x187360e68 __NSThread__start__ + 1024
7  libsystem_pthread.dylib        0x1858e1850 <redacted> + 240
8  libsystem_pthread.dylib        0x1858e1760 _pthread_start + 282
9  libsystem_pthread.dylib        0x1858ded94 thread_start + 4
```

Any ideas?

**Comment by patchthecode** [comment]
I have fixed some issues since version 6.1.2
The latest version is 6.1.5

You can see a record of the crash fixes here on the change log. -> https://github.com/patchthecode/JTAppleCalendar/blob/master/CHANGELOG.md

Unfortunately I cannot tell you much from the log, but I can tell you that the latest versions are more stable than 6.1.2 as seen the change logs.

Are you doing anything "fancy" on your calendar?

**Comment by jdmarshall90** [comment]
I don't know about fancy, but my calendar's cells have an uneditable text field with a custom input view.

**Comment by patchthecode** [comment]
I Have closed this issue but attached label: PROD Crash for version 6.1.2
Issue was closed since many fixed have been done since version 6.1.2
I believe that it is fixed in later versions.

However, I will monitor any further prod crash reports from other people to see if the error was truly fixed.

You were the 1st Prod crash i have recorded.
